overlord/devicestate/handlers_gadget.go

```diff
@@ -212,3 +213,115 @@ func (m *DeviceManager) doUpdateGadgetAssets(t *state.Task, _ *tomb.Tomb) error
 
 	return nil
 }
+
+func (m *DeviceManager) updateGadgetCommandLine(t *state.Task, st *state.State, isUndo bool) (updated bool, err error) {
+	snapsup, err := snapstate.TaskSnapSetup(t)
+	if err != nil {
+		return false, err
+	}
+	devCtx, err := DeviceCtx(st, t, nil)
+	if err != nil {
+		return false, err
+	}
+	if devCtx.Model().Grade() == asserts.ModelGradeUnset {
+		// pre UC20 system, do nothing
+		return false, nil
+	}
+	// but when undoing, we use the current gadget which should have
```

Comment:
Hm, this comment starts with `// but ...` - what does the "but" refers to? Should this maybe read just "when undoing, we use the current gadget which should have been restored"

Reply:
Could this move into the "else" below? It seems we only need currentGadgetData inside this else condition?